src/kademlia/node.cpp

```diff
@@ -565,6 +566,39 @@ void node::put_item(public_key const& pk, std::string const& salt
 	ta->start();
 }
 
+void node::sample_infohashes(udp::endpoint const& ep, sha1_hash const& target
+	, std::function<void(time_duration
+		, int, std::vector<sha1_hash>
+		, std::vector<std::pair<sha1_hash, udp::endpoint>>)> f)
+{
+#ifndef TORRENT_DISABLE_LOGGING
+	if (m_observer != nullptr && m_observer->should_log(dht_logger::node))
+	{
+		m_observer->log(dht_logger::node, "starting sample_infohashes for [ node: %s, target: %s ]"
+			, print_endpoint(ep).c_str(), aux::to_hex(target).c_str());
+	}
+#endif
+
+	// not an actual traversal
+	auto ta = std::make_shared<dht::sample_infohashes>(*this, node_id(), std::move(f));
+
+	auto o = m_rpc.allocate_observer<sample_infohashes_observer>(ta, ep, node_id());
+	if (!o) return;
+#if TORRENT_USE_ASSERTS
+	o->m_in_constructor = false;
+#endif
+
+	entry e;
+	entry& a = e["a"];
+
+	e["q"] = "sample_infohashes";
+	a["target"] = target.to_string();
```

Comment:
you could simplify this a bit by saying: ``e["a"]["target"] = target.to_string()``. In fact, I don't believe you need to say ``to_string()``, since target will also fit the ``span`` overload (unless I'm mistaken), which may also avoid the heap allocation.